app/common/test/controllers/labRequestController.js

```diff
@@ -195,6 +195,28 @@
     function showTestOrderDetails(testOrder) {
       vm.testsOfSelectedRequest = testOrder.testOrderItems;
       vm.testOrderInDetail = testOrder;
+      testOrderResultService.getTestOrderResult(testOrder.uuid).then(function (testOrderResult) {
+        var itemResults = testOrderResult.items;
+        itemResults.forEach(function (itemResult) {
+          var currentItemResult = vm.testsOfSelectedRequest.find(function (testItem) {
```

Comment:
I would suggest that the this logic be implemented as a single method call to a service. It's not required but from experience it gets really hard to refactor when we assign values to controller properties inside nested callbacks.